include/loops/loops.hpp

```diff
@@ -271,6 +259,7 @@ class Context
     std::string getPlatformName() const;
     void compileAll();
 protected:
+    Context(Context* a_impl): impl(a_impl) {}
```

Comment:
The problem is that ContextImpl have to directly access FuncImpl's methods, which are not presented in Func interface[hidden] and vice versa(at some point there was also Backend, but even two participants is enough). Therefore, ContextImpl and FuncImpl must be public symbols. On the other hand we don't want to show these details to user. Now user even don't have access to name of Imp

Since they are defined in loops namespace(means - global), they can access each others public methods, and these methods are not